sycl/test/scheduler/MemObjRemapping.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=1 %CPU_RUN_PLACEHOLDER %t.out 2>&1 %CPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=1 %ACC_RUN_PLACEHOLDER %t.out 2>&1 %ACC_CHECK_PLACEHOLDER
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <cstddef>
+
+using namespace cl::sycl;
+
+class Foo;
+class Bar;
+
+// This test checks that memory objects are remapped on requesting an access mode
+// incompatible with the current mapping.
+int main() {
+  queue Q;
+
+  std::size_t Size = 1024;
+  range<1> Range{Size};
+  buffer<int, 1> Buf{Range};
+
+  Q.submit([&](handler &Cgh) {
+    auto Acc = Buf.get_access<access::mode::read_write>(Cgh);
+    Cgh.parallel_for<Foo>(Range, [=](id<1> Idx) {
+      Acc[Idx] = Idx[0];
+    });
+  });
+
+  {
+    // CHECK: piEnqueueMemBufferMap
```

Comment:
It might be hard to obtain and check the appropriate flags here, but without them it's also hard to say if the feature works correctly.

The code itself in the sources is LGTM.

Reply:
Fair enough. Do you think checking the raw value of flags would be acceptable here?

Reply:
It would be fine for me.